[BUG] SIGSEGV when validating workflow without author, silent error when workflow does not contain author

### Is there an existing issue for this?

- [x] I have searched the existing issues.

### Current Behavior

When running it like this we get SIGSEGV
```./nuclei -u xxx.apple.com -silent -jsonl -duc -code -timeout 60 -ud templates/nuclei/ -w "templates/nuclei/host-based-workflows/xxx.apple.com/workflow.yaml" -validate
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1c0 pc=0x206f7fd]

goroutine 1 [running]:
github.com/projectdiscovery/nuclei/v3/pkg/catalog/loader.(*Store).areWorkflowOrTemplatesValid(0xc00076d720, 0xc000dcfd08?, 0x1, 0x357c248)
	github.com/projectdiscovery/nuclei/v3/pkg/catalog/loader/loader.go:368 +0x39d
github.com/projectdiscovery/nuclei/v3/pkg/catalog/loader.(*Store).areWorkflowsValid(...)
	github.com/projectdiscovery/nuclei/v3/pkg/catalog/loader/loader.go:327
github.com/projectdiscovery/nuclei/v3/pkg/catalog/loader.(*Store).ValidateTemplates(0xc00076d720)
	github.com/projectdiscovery/nuclei/v3/pkg/catalog/loader/loader.go:320 +0x374
github.com/projectdiscovery/nuclei/v3/internal/runner.(*Runner).RunEnumeration(0xc00082c0f0)
	github.com/projectdiscovery/nuclei/v3/internal/runner/runner.go:557 +0x1534
main.main()
	./main.go:184 +0x712
```

the following template can be used as test case:
```
id: xxx-apple-com-workflow
info:
  name: Specific workflow of xxx.com hostname
  description: Workflow for xxx.com


workflows:
# - template: ./general/
# - template: ./general-fe/
# - template: ./host-based-workflows/xxx.apple.com/adhoc/
- template: ./host-based-workflows/xxx.apple.com/regression/
```
Silent error also happened [FTL] Could not run nuclei: no templates provided for scan


### Expected Behavior

no SIGSEGV

### Steps To Reproduce

./nuclei -u xxx.apple.com -silent -jsonl -duc -code -timeout 60 -ud templates/nuclei/ -w "templates/nuclei/host-based-workflows/xxx.apple.com/workflow.yaml" -validate

### Relevant log output

_No response_

### Environment

- OS: ubuntu
- Nuclei: latest
- Go: - (binary version of nuclei)


### Anything else?

_No response_

--- a/pkg/catalog/loader/loader.go
+++ b/pkg/catalog/loader/loader.go
@@ -352,6 +352,7 @@ func (store *Store) areWorkflowOrTemplatesValid(filteredTemplatePaths map[string
 		if err != nil {
 			if isParsingError("Error occurred parsing template %s: %s\n", templatePath, err) {
 				areTemplatesValid = false
+				continue
 			}
 		} else if template == nil {
 			// NOTE(dwisiswant0): possibly global matchers template.

